packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/method_channel_multi_factor.dart

```diff
@@ -178,6 +210,104 @@ class MethodChannelPhoneMultiFactorGenerator
   MultiFactorAssertionPlatform getAssertion(
     PhoneAuthCredential credential,
   ) {
-    return MultiFactorAssertion(credential);
+    return PhoneMultiFactorAssertion(credential);
+  }
+}
+
+class TotpMultiFactorAssertion extends MultiFactorAssertion {
+  TotpMultiFactorAssertion(
+    this.assertionId,
+  ) : super(null);
+
+  final String assertionId;
+}
+
+/// Helper class used to generate PhoneMultiFactorAssertions.
+class MethodChannelTotpMultiFactorGenerator
+    extends TotpMultiFactorGeneratorPlatform {
+  final _api = MultiFactorTotpHostApi();
+
+  /// Transforms a PhoneAuthCredential into a [MultiFactorAssertion]
+  /// which can be used to confirm ownership of a phone second factor.
+  @override
+  Future<TotpSecretPlatform> generateSecret(
+    MultiFactorSession session,
+  ) async {
+    final pigeonSecret = await _api.generateSecret(session.id);
+    return MethodChannelTotpSecret(
+      pigeonSecret.codeIntervalSeconds,
+      pigeonSecret.codeLength,
+      pigeonSecret.enrollmentCompletionDeadline != null
+          ? DateTime.fromMillisecondsSinceEpoch(
+              pigeonSecret.enrollmentCompletionDeadline!,
+            )
+          : null,
+      pigeonSecret.hashingAlgorithm,
+      pigeonSecret.secretKey,
+    );
+  }
+
+  /// Get a [MultiFactorAssertion]
+  /// which can be used to confirm ownership of a TOTP second factor.
+  @override
+  Future<MultiFactorAssertionPlatform> getAssertionForEnrollment(
+    TotpSecretPlatform secret,
+    String oneTimePassword,
+  ) async {
+    final totpAssertionId =
+        await _api.getAssertionForEnrollment(secret.secretKey, oneTimePassword);
+    return TotpMultiFactorAssertion(totpAssertionId);
+  }
+
+  /// Get a [MultiFactorAssertion]
+  /// which can be used to confirm ownership of a TOTP second factor.
+  @override
+  Future<MultiFactorAssertionPlatform> getAssertionForSignIn(
+    String enrollmentId,
+    String oneTimePassword,
+  ) async {
+    final totpAssertionId =
+        await _api.getAssertionForSignIn(enrollmentId, oneTimePassword);
+    return TotpMultiFactorAssertion(totpAssertionId);
+  }
+}
+
+/// Helper class used to generate PhoneMultiFactorAssertions.
+class MethodChannelTotpSecret extends TotpSecretPlatform {
+  MethodChannelTotpSecret(
+    super.codeIntervalSeconds,
+    super.codeLength,
+    super.enrollmentCompletionDeadline,
+    super.hashingAlgorithm,
+    super.secretKey,
+  );
+
+  final _api = MultiFactorTotpSecretHostApi();
+
+  /// Returns a QR code URL as described in https://github.com/google/google-authenticator/wiki/Key-Uri-Format
+  /// This can be displayed to the user as a QR code to be scanned into a TOTP app like Google Authenticator.
+  /// If the optional parameters are unspecified, an accountName of and issuer of are used.
```

Comment:
Are the default values of `accountName` and `issuer` known? If so, can we update the doc here?